src/CoreBundle/private/library/classes/TFullGroupTable/TFullGroupTable.class.php

```diff
@@ -866,10 +921,22 @@ protected function _BuildFilterSection()
 
         if ($this->showSearchBox) {
             $this->somethingToShow = true;
+
+            $filterContent .= '<div class="form-group mr-2">';
+            $filterContent .= '<select id="searchLookup" name="_search_word" data-listname="'. TGlobal::OutHTML($this->listName) .'" class="form-control" data-select2-placeholder="';
+            $filterContent .= $this->searchFieldText;
+            $filterContent .= '" data-select2-ajax="'. $this->getAjaxRecordsUrl() .'">';
+
+            if ('' !== $this->_postData['_search_word']) {
+                $filterContent .= '<option value="'.TGlobal::OutHTML($this->_postData['_search_word']).'" selected>'.TGlobal::OutHTML($this->_postData['_search_word']).'</option>';
+            }
+            $filterContent .= '</select></div>';
+
             $filterContent .= '<div class="form-group">';
-            $filterContent .= '<input type="text" class="form-control form-control-sm" name="_search_word" value="'.TGlobal::OutHTML($this->_postData['_search_word'])."\" onKeyDown=\"if (window.event && window.event.keyCode == 13) document.{$this->listName}._startRecord.value=0\" onChange=\"document.{$this->listName}._startRecord.value=0;document.{$this->listName}.submit();\" placeholder=\"".$this->searchFieldText."\">\n";
-            $filterContent .= "<input type=\"button\" class=\"form-control form-control-sm btn-sm btn-primary\" value=\"{$this->searchButtonText}\" onClick=\"document.{$this->listName}._startRecord.value=0;document.{$this->listName}.submit();\" class=\"btn btn-sm btn-primary\">
-            </div>";
+            $filterContent .= '<input type="button" class="form-control form-control-sm btn-sm btn-primary" value="'.$this->searchButtonText.'" onClick="document.'.$this->listName.'._startRecord.value=0;document.'.$this->listName.'.submit();" class="btn btn-sm btn-primary">';
```

Comment:
`$this->searchButtonText` needs to be escaped.

Reply:
Done.